neo/Settings.py

```diff
@@ -32,6 +37,52 @@
 FILENAME_SETTINGS_COZNET = os.path.join(DIR_PROJECT_ROOT, 'protocol.coz.json')
 
 
+class PrivnetConnectionError(Exception):
+    pass
+
+
+class DependencyError(Exception):
+    pass
+
+
+def check_privatenet():
+    """ Check if privatenet is running, and if container is same as the chain file """
+    rpc_settings.setup(settings.RPC_LIST)
+    client = RPCClient()
+    version = client.get_version()
+    if not version:
+        raise PrivnetConnectionError("Error: private network container doesn't seem to be running, or RPC is not enabled.")
+
+    print("Privatenet useragent '%s', nonce: %s" % (version["useragent"], version["nonce"]))
+
+    # Now check if nonce is the same as in the chain path
+    nonce_container = str(version["nonce"])
+    neopy_chain_meta_filename = os.path.join(settings.LEVELDB_PATH, ".privnet-nonce")
+    if os.path.isfile(neopy_chain_meta_filename):
+        nonce_chain = open(neopy_chain_meta_filename, "r").read()
+        if nonce_chain != nonce_container:
+            raise PrivnetConnectionError(
+                "Chain database in Chains/privnet is for a different private network than the current container. "
+                "Consider deleting the Chain directory with 'rm -rf Chains/privnet*'."
+            )
+    else:
+        with open(neopy_chain_meta_filename, "w") as f:
+            f.write(nonce_container)
+
+
+def check_depdendencies():
+    # Get installed packages
+    installed_packages = pip.get_installed_distributions()
+    installed_packages_list = sorted(["%s==%s" % (i.key, i.version) for i in installed_packages])
+
+    # Now check if each package specified in requirements.txt is actually installed
+    deps_filename = os.path.join(DIR_PROJECT_ROOT, "requirements.txt")
+    deps_list = open(deps_filename, "r").read().split()
+    for dep in deps_list:
+        if not dep.lower() in installed_packages_list:
+            raise DependencyError("Required dependency %s is not installed. Please run 'pip install -e .'." % dep)
```

Comment:
Don't forget to close the file to free up the system resources (even if it's just a small file). Ideally use `with open` because that will make sure to close the file even if for some reason an exception is raised.

Reply:
thanks! downside is it adds 2 lines :P

Reply:
updated